### PR TITLE
fix: typescript clien gen conflict with module

### PR DESCRIPTION
**Pull request by TomChv**
I noticed during a demo that if a module source is `.` and  implements functions and also generate a client, the `sdk/index.ts` file will not expose the generated client in the `sdk` folder because it doesn't expect it to be there.

I changes the logic in the runtime to handle this specific edge case and updated integrations tests.

This behaviour could be handled in the codegen binary ideally but it's a quick fixes for now.

### CODE REVIEW

File: core/integration/client_generator_test.go [hunk 1]
```diff
@@ -476,11 +478,11 @@ export class Test {
...
 async function main() {
-    await connection(async () => {
-      const res = await dag.test().hello()
+  await connection(async () => {
+    const res = await dag.test().hello()
 
-      console.log("result:", res)
-    })
+    console.log("result:", res)
+  })
```
Comment (cwlbraa) [inline core/integration/client_generator_test.go:485]: nit: fix your editor's indent settings

**Review by cwlbraa — APPROVED** [review]

File: core/integration/client_generator_test.go (hunk 1 above)
Comment (TomChv) [reply core/integration/client_generator_test.go:485]: The string snippet are not handled by my editor sadly haha so I have to do it manually every time! Good catch

Comment (TomChv) [reply core/integration/client_generator_test.go:485]: Here I'm actually fixing a indent issue from a previous PR, normally we use 2 space, not 4 :)

Comment (cwlbraa) [reply core/integration/client_generator_test.go:485]: nit: fix someone else's editor indent settings lol
